### PR TITLE
[Windows] add generated plugins ignores

### PR DESCRIPTION
**Pull request by hlwhl**
Add generated plugins ignores for windows platform.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [ ] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -107,6 +107,10 @@ unlinked_spec.ds
 **/macos/Flutter/ephemeral
 **/xcuserdata/
 
+# Windows
+**/windows/flutter/generated_plugins.cmake
+**/windows/flutter/generated_plugin_registrant.cc
```
Comment (loic-sharma) [inline .gitignore:112]: Should this include the header file?

```suggestion
**/windows/flutter/generated_plugin_registrant.cc
**/windows/flutter/generated_plugin_registrant.h
```

Comment (hlwhl) [reply .gitignore:112]: Yes, the header file added!

**Review by loic-sharma — APPROVED** [review]
LGTM

**Comment by loic-sharma** [comment]
Any concerns @cbracken?

It looks like you may need to rebase off the lastest `master` for the CI to pass

**Comment by hlwhl** [comment]
> Any concerns @cbracken?
> 
> It looks like you may need to rebase off the lastest `master` for the CI to pass

Rebased and CI passed.

File: .gitignore [hunk 2]
```diff
@@ -107,6 +107,11 @@ unlinked_spec.ds
 **/macos/Flutter/ephemeral
 **/xcuserdata/
 
+# Windows
+**/windows/flutter/generated_plugins.cmake
```
Comment (loic-sharma) [inline .gitignore:111]: Turns out this should be checked-in. See: https://github.com/flutter/flutter/pull/110114#discussion_r955500880

Comment (loic-sharma) [inline .gitignore:111]: Turns out this should be checked-in. Apologies for the churn! See: https://github.com/flutter/flutter/pull/110114#discussion_r955500880

```suggestion
```

Comment (hlwhl) [reply .gitignore]: No problem, thanks for the clarification!

**Review by cbracken — APPROVED** [review]
LGTM stamp from a Japanese personal seal

lgtm thanks -- these files are ephemeral and shouldn't be checked in; thanks for fixing the issue. This will need a test exemption from @Hixie.

**Comment by Hixie** [comment]
test-exempt: configuration change

**Comment by loic-sharma** [comment]
Thanks for the contribution! 🎉

**Pull request merged**